src/frontend/screens/ObservationEdit/LocationView.tsx

```diff
@@ -19,66 +19,95 @@ const m = defineMessages({
 });
 
 export const LocationView = () => {
-  const liveLocation = useMostAccurateLocationForObservation();
-  const observationValue = usePersistedDraftObservation(
-    observationValueSelector,
+  const observationValue = usePersistedDraftObservation(store => store.value);
+
+  return observationValue?.metadata.manualLocation ? (
+    <LocationViewManualPosition />
+  ) : (
+    <LocationViewUpdatePosition />
+  );
+};
+
+const LocationViewManualPosition = () => {
+  const observationValue = usePersistedDraftObservation(store => store.value);
+  return (
+    <LocationViewInner
+      lat={observationValue?.lat}
+      lon={observationValue?.lon}
+      accuracy={null}
+    />
   );
-  const coordinateFormat = usePersistedSettings(coordinateFormatSelector);
+};
+
+const LocationViewUpdatePosition = () => {
+  const observationValue = usePersistedDraftObservation(store => store.value);
+  //only update the location if the accuracy increases OR the user has moved outside of the accuracy's radius of uncertainty.
+  const {location} = useLocation(
+    ({accuracy}) => accuracy === 'better' || accuracy === 'newBounds',
+  );
+  const {updateObservationPosition} = useDraftObservation();
+
+  useEffect(() => {
+    const newCoord = !location
+      ? undefined
+      : Object.entries(location.coords).map(
+          ([key, val]) => [key, val === null ? undefined : val] as const,
+        );
 
-  const coordinateInfo = observationValue?.metadata.manualLocation
-    ? {
-        lat: observationValue.lat,
-        lon: observationValue.lon,
-        accuracy: liveLocation?.coords?.accuracy,
-      }
-    : {
-        lat: liveLocation?.coords?.latitude,
-        lon: liveLocation?.coords?.longitude,
-        accuracy: liveLocation?.coords?.accuracy,
-      };
+    updateObservationPosition({
```

Comment:
I think updating the observation location just based on whether there has been a new location update, and relying on renders from the hook, seems possibly flakey, and for a future maintainer reading this there is some indirection to find the logic. Consider moving the logic here, and comparing with the accuracy of the location saved to the observation, rather than the accuracy change in the location state.

Reply:
A problem with that is the hook will cause a new re-render on every change, even if we are not updating the UI. Because we would not be filtering the accuracy, it would received a new state value on every accuracy change but we would still keep the UI the same if the accuracy didn't increase.

Reply:
Will be addressed in [this](https://github.com/digidem/comapeo-mobile/issues/402)